core/src/modules/ops/destination.ts

```diff
@@ -912,82 +922,115 @@ export namespace DestinationOps {
     success: boolean;
     error: Error;
   }> {
-    const _exports: Export[] = []; // only ones we are sending
-    for (const _export of givenExports) {
-      // TODO: maybe should recalcuate hasChanges and from current mostRecent
-      if (!_export.hasChanges) {
-        await _export.complete(); // do not include exports with hasChanges=false
-      } else {
-        _exports.push(_export);
-      }
-    }
+    const _exports: Export[] = [];
+    const locks: Awaited<ReturnType<typeof getLock>>[] = [];
 
-    const exportRecords: ExportRecordsPluginMethod = await getBatchFunction(
-      destination
-    );
+    try {
+      for (const givenExport of givenExports) {
+        if (!givenExport.hasChanges) {
+          await givenExport.complete();
+          continue;
+        }
 
-    const syncMode = await destination.getSyncMode();
-    const syncOperations: DestinationSyncOperations = syncMode
-      ? DestinationSyncModeData[syncMode].operations
-      : DestinationSyncModeData.sync.operations; // if destination does not support sync modes, allow all
+        const mostRecentExport = await Export.findOne({
```

Comment:
Calling out that this will, in the short term, slow down this method because we need to SELECT 100 times now (once for each export).  Once we do https://www.pivotaltracker.com/story/show/181115564, we can skip this lookup